libsoftwarecontainer/src/gateway/network/networkgateway.cpp

```diff
@@ -172,25 +172,28 @@ ReturnCode NetworkGateway::down()
         Netlink n;
         Netlink::LinkInfo iface;
         if (isError(n.findLink("eth0", iface))) {
-            log_error() << "Could not find interface eth0 in container";
-            return 1;
+            return NO_LINK;
         }
 
         if (isError(n.linkDown(iface.first.ifi_index))) {
-            log_error() << "Could not bring interface eth0 down in container";
-            return 2;
+            return BAD_LINKUPDOWN;
         }
 
-        return 0;
+        return SUCCESS;
     });
     job.start();
-    job.wait();
-    if (job.isError()) {
-        log_error() << "Configuring eth0 to 'down state' failed.";
-        return ReturnCode::FAILURE;
+    int returnCode = job.wait();
+    switch(returnCode)
+    {
+        case NO_LINK:
+            log_error() << "Could not find interface eth0 in container";
+            return ReturnCode::FAILURE;
+        case BAD_LINKUPDOWN:
+            log_error() << "Could not bring interface eth0 down in container";
+            return ReturnCode::FAILURE;
+        default:
```

Comment:
maybe you can add `case FAILURE: ` for more readable code or instead default adding `case SUCCESS`